### DISCUSSION TITLE
Bitmark DID Format

### DISCUSSION THREAD

**Discussion by nodestory** [bitmark-inc/bitmark-protocol#18] [Ideas]
The current design is as follows.

```
bmk-did        = "did:bmk:" bmk-identifier
                 [ ";" did-service ] [ "/" did-path ]
                 [ "?" did-query ] [ "#" did-fragment ]
bmk-identifier = [chain][version][SHA3-512(deidentified-did-doc)]
```

### How to handle updates of IID docs?

By the current approach, when the registrant wants to update the IID document, the identifier will change.
Almost every field in the IID document for a FF asset can be updated afterwards.

**Existing use cases:**
- Updates to the original artwork
  - The artist laster found that his 3D artwork is not printable. He has to remove a certain layer from the model to fix the problem.
  - An AR artwork cannot be correctly displayed on all browsers. The artist wants to apply a patch to the design.
  → This is similar to _editions_ of a book, and the new edition should get a new ISBN number. Do we consider the updated version as the original asset?

- Updates to metadata
  - An artist wants to append the creation date of the artwork, which is not included in the asset metadata.

**Possible use cases**
- Updates to the art sale agreement
  - We might want to update the revenue sharing rule.

### Shall we distinguish assets and titles from identifier?

From [`did:git`](https://github.com/dhuseby/did-git-spec/blob/master/did-git-spec.md#namespace-specific-identifier-nsi)(though the spec is already withdrawn), there are two valid DID formats:

1. The DID for just the repo: `did:git:<Repo ID>`.
2. The DID for a contributor identity in the repo: `did:git:<Repo ID>:<Contributor ID>`.

The Bitmark method can take the same formats:

1. The DID for the asset: `did:bmk:<asset-id>`.
2. The DID for the title: `did:bmk:<asset-id>:<edition>`.

**Comment by hxw** [comment]
Questions/Comments?
- Why do almost every field need to be mutable? e.g. can the artist rename the artwork.
- Why should updates change the ID? e.g. transfer ownership should not.
- Second editions of books are not usually free, so a second issue even if exactly the same must have a new id.
- I would expect most field to be immutable so any "patching" would be a new ID. I can see the case for edition 1of a first "printing" being very valuable and if the artwork can be arbitrarily changed that value could be negated.

**Reply by nodestory** [reply]
If any of the linked resources are updated, the IID would change.

Probably the book edition is not a good analogy. When artists apply a patch, what FF currently wants is that all collectors can get access to the latest patch, which is similar to a software patch.

If the updated asset is treated as a new asset, that means we have to issue new titles and transfer them to current collectors?
